### PR TITLE
Added JUnitAdapter

### PR DESCRIPTION
**Pull request by Gyanesha**
I wrote the JUnitAdapter and generated a random serial version UID for this.

### CODE REVIEW

**Comment by uhafner** [comment]
Hmm, the file is empty... can you please check?

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101&bid=15985180)

**Comment by Gyanesha** [comment]
I am sorry to ask such a question but what else changes I have to make in order to add the new file as a part of the project ?

**Comment by uhafner** [comment]
Please increase the version number in the pom to 7.1.0-SNAPSHOT. New classes are not allowed in patch releases.

Can you please add a test case as well. I'm not sure what you are expecting to get as result for failed, skipped, and successful tests. 

Please also make sure that there is a JavaDoc at class level (and remove superfluous empty lines).

File: src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package edu.hm.hafner.analysis.parser.violations;
+
+import se.bjurr.violations.lib.parsers.JUnitParser;
+
+public class JUnitAdapter extends AbstractViolationAdapter {
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java]: JavaDoc.

File: src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    private static final long serialVersionUID = -1595503635554896281L;
+
+    @Override
+    protected JUnitParser createParser() {
+        return new JUnitParser();
+    }
+
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java]: remove new lines

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101&bid=15985180)

**Comment by Gyanesha** [comment]
> Please increase the version number in the pom to 7.1.0-SNAPSHOT. New classes are not allowed in patch releases.
> 
> Can you please add a test case as well. I'm not sure what you are expecting to get as result for failed, skipped, and successful tests.
> 
> Please also make sure that there is a JavaDoc at class level (and remove superfluous empty lines).

**Comment by Gyanesha** [comment]
Sir, in order to add JUnit Parser I have made the class JUnitAdapter which causes breaking of semantic versioning. How can I avoid this? Or where in which class I have to add JUnitAdapter.

**Comment by uhafner** [comment]
I have already prepared the next release version as 8.0.0 in master. I.e. you need to rebase your changes on master (or merge with master). Then you get no errors locally (the build already passes on CI).

File: src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package edu.hm.hafner.analysis.parser.violations;
+
+import se.bjurr.violations.lib.parsers.JUnitParser;
+
+public class JUnitAdapter extends AbstractViolationAdapter {
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java]: Codacy Issue found: [Missing a Javadoc comment.](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101)

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101&bid=15985180)

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest1.java  1
- src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest2.java  1
- src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101&bid=15985180)

File: src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java (hunk 3 above)
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java]: Codacy Issue found: [Missing a Javadoc comment.](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101)

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest1.java  1
- src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest2.java  1
- src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101&bid=15985180)

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest1.java  1
- src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest2.java  1
- src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101&bid=15985180)

File: src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest2.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package edu.hm.hafner.analysis.parser.violations;
+
+import edu.hm.hafner.analysis.AbstractParserTest;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+import edu.hm.hafner.analysis.assertj.SoftAssertions;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class JUnitAdapterTest2 extends AbstractParserTest {
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest2.java]: How is this test different from the previous one? If it is really different, then please add it as a second test case to class `JunitAdapterTest` and describe the difference in the JavaDoc.:
```
    @Test
    void should***MeaningfulName****() {
        Report report = parse("filename.txt");

        assertThat(report).hasSize(1);
    }

```

File: src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    @Override
+    protected JUnitAdapter createParser() {
+        return new JUnitAdapter();
+    }
+
+    /** Verifies that violations can be parsed from JUnit2 */
+    @Test
+    void shouldParseWithJUnit2(){
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest.java]: Codacy Issue found: ['{' is not preceded with whitespace.](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101)

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest.java  1
- src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101&bid=15985180)

File: src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+                .contains("java.lang.AssertionError");
+
+    }
+
+    @Override
+    protected JUnitAdapter createParser() {
+        return new JUnitAdapter();
+    }
+
+    /** Verifies that violations can be parsed from JUnit2 */
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest.java]: Codacy Issue found: [First sentence should end with a period.](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101)

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest.java  1
- src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101&bid=15985180)

File: src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    @Override
+    protected JUnitAdapter createParser() {
+        return new JUnitAdapter();
+    }
+
+    /**
+     * Verifies that violations can be parsed from JUnit2.
+     * */
+    @Test
+    void shouldParseWithJUnit2(){
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest.java]: Codacy Issue found: ['{' is not preceded with whitespace.](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101)

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest.java  1
- src/main/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=4804101&bid=15985180)

File: src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest2.java (hunk 4 above)
Comment (Gyanesha) [reply src/test/java/edu/hm/hafner/analysis/parser/violations/JUnitAdapterTest2.java]: I have added the second test case, according to what you instructed me.

**Comment by uhafner** [comment]
Thanks!

**Comment by Gyanesha** [comment]
Thank you Sir for guiding me to make this pull request.

**Comment by uhafner** [comment]
Released in https://github.com/jenkinsci/analysis-model/releases/tag/analysis-model-8.0.0-beta5.

**Pull request merged**